config/dependency.go

```diff
@@ -25,15 +26,20 @@ type Dependency struct {
 // Decode the dependency blocks from the file, and then retrieve all the outputs from the remote state. Then encode the
 // resulting map as a cty.Value object.
 // TODO: In the future, consider allowing importing dependency blocks from included config
+// NOTE FOR MAINTAINER: When implementing importation of other config blocks (e.g referencing inputs), carefully
+//                      consider whether or not the implementation of the cyclic dependency detection still makes sense.
```

Comment:
Depending on how this is implemented, what we have now may or may not work. E.g if someone specifies `skip_output` to `true`, should that dependency be an explicit dependency of the module such that it affects the execution order? If no, then the current implementation works because it will exclude that dependency from the dependency list.